crud/common/sharding/sharding_func.lua

```diff
@@ -0,0 +1,88 @@
+local errors = require('errors')
+
+local dev_checks = require('crud.common.dev_checks')
+
+local ShardingFuncError = errors.new_class('ShardingFuncError',  {capture_stack = false})
+
+local sharding_func_module = {}
+
+local function get_function_from_G(func_name)
+    local chunks = string.split(func_name, '.')
+    local sharding_func = _G
+
+    for _, chunk in pairs(chunks) do
+        if sharding_func == nil then
+            return nil
+        end
+
+        sharding_func = rawget(sharding_func, chunk)
+    end
+
+    return sharding_func
+end
+
+local function as_callable_object(sharding_func_def)
+    if type(sharding_func_def) == 'string' then
+        local sharding_func = get_function_from_G(sharding_func_def)
+        if sharding_func == nil then
+            return nil, ShardingFuncError:new(
+                    "No sharding func name (%s) in _G", sharding_func_def)
+        end
+
+        return sharding_func
+    end
+
+    if type(sharding_func_def) == 'table' then
+        local sharding_func, err = loadstring('return ' .. sharding_func_def.body)
+        if sharding_func == nil then
+            return nil, ShardingFuncError:new(
+                    "Incorrect sharding func body: %s", err)
+        end
+
+        return sharding_func()
+    end
+
+    return nil, ShardingFuncError:new(
+            "Wrong sharding function type")
+end
+
+function sharding_func_module.extract_def_from_tuple(tuple)
```

Comment:
The function name confused me. 'tuple' in name is not a tuple with data, it is a tuple from `box.space._ddl_sharding_func` where we store sharding metadata (sharding keys and functions).
I would rename the function to something like `extract_function_def`.

Reply:
Renamed:

```lua
function sharding_func_module.extract_function_def(tuple)
    if not tuple then
        return nil
    end
...
end
```